modules/openapi-generator/src/main/resources/cpp-restbed-server/api-source.mustache

```diff
@@ -48,6 +48,10 @@ void {{classname}}::stopService() {
 		std::bind(&{{classname}}{{vendorExtensions.x-codegen-resourceName}}Resource::{{httpMethod}}_method_handler, this,
 			std::placeholders::_1));
 	{{/vendorExtensions.x-codegen-otherMethods}}
+
+	{{#allParams}}{{#defaultValue}}{{^isListContainer}}{{paramName}} = {{{defaultValue}}};
+	{{/isListContainer}}{{/defaultValue}}{{^defaultValue}}{{#isModel}}{{paramName}} = std::make_shared<{{baseType}}>();
```

Comment:
Is there a reason to make models `std::shared_ptr`? This will cause problems with using it with no observable benefit (at least I don't see any).

Consider following yaml:
```yaml 
openapi: 3.0.0
info:
  description: Some description
  version: 0.0.1
  title: Some title

tags:
  - name: hello

paths:
  "/there":
    get:
      operationId: helloThereGet
      tags:
        - hello
      summary: Do something
      responses:
        200:
          description: Successful operation
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/OurModel"
    post:
      requestBody:
        description: json to deserialize then serialize
        required: true
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/OurModel"
      operationId: helloTherePost
      tags:
        - hello
      summary: Do something (receive)
      responses:
        200:
          description: Successful operation
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/OurModel"
servers:
  - url: http://localhost:8080
components:
  schemas:
    OurModel:
      type: object
      properties:
        someints:
          type: array
          items:
            type: integer
        member:
          $ref: '#/components/schemas/NestedMember'
        strmem:
          type: string
    NestedMember:
      type: object
      properties:
        number:
          type: integer
        text:
          type: string
      required:
        - number
        - text
```

test file:
```cpp
#include <gtest/gtest.h>
#include "model/OurModel.h"

TEST(OurModel, SimpleTest)
{
  using namespace org::openapitools::server::model;

  OurModel model;
  model.setSomeints({1, 2, 3});
  
  auto nm = std::make_shared<NestedMember>();
  nm->setNumber(10);
  nm->setText("some text");

  model.setMember(nm);

  EXPECT_EQ((std::vector<int32_t>{1, 2, 3}), model.getSomeints()); // success
  EXPECT_EQ(10, model.getMember()->getNumber()); // success
  EXPECT_EQ("some text", model.getMember()->getText()); // success

  OurModel copy = model;
  copy.setSomeints({4, 5, 6});
  copy.getMember()->setText("this modifies model->member->text too");

  EXPECT_EQ((std::vector<int32_t>{1, 2, 3}), model.getSomeints()); // success
  EXPECT_EQ(10, model.getMember()->getNumber()); // success
  EXPECT_EQ("some text", model.getMember()->getText()); // fail, even though it got modified in other object

  EXPECT_EQ((std::vector<int32_t>{4, 5, 6}), copy.getSomeints()); // success
  EXPECT_EQ(10, copy.getMember()->getNumber()); // success
  EXPECT_EQ("this modifies model->member->text too", copy.getMember()->getText()); // success
}
```

Output:
```
main.cpp:27: Failure
Expected equality of these values:
  "some text"
  model.getMember()->getText()
    Which is: "this modifies model->member->text too"
```

I'd be fine with using `shared_ptr` if every property would use it. But it's inconsistent, you can copy your model's object, modify any property except for nested models and it won't affect original object. Then you would modify nested object and it will modify any number of other objects, which probably isn't what we want.

Reply:
Might just be due to lack of experience. I had this intentionally after it would not compile with my testing changes, seems to work just fine without. I've removed the lines in the most recent commit!